packages/server/graphql/queries/helpers/repoIntegrationHelpers.ts

```diff
@@ -0,0 +1,77 @@
+import ms from 'ms'
+import {Unpromise} from 'parabol-client/types/generics'
+import makeRepoIntegrationId from 'parabol-client/utils/makeRepoIntegrationId'
+import getRethink from '../../../database/rethinkDriver'
+import {DataLoaderWorker} from '../../graphql'
+
+export interface IntegrationByTeamId {
+  id: string
+  userId: string
+  service: 'github' | 'jira'
+  nameWithOwner: string | null
+  projectKey: string | null
+  projectName: string | null
+  avatar: string | null
+  cloudId: string | null
+  lastUsedAt: Date
+}
+
+export const getUserIntegrationIds = async (
+  userId: string,
+  teamId: string,
+  permLookup: Unpromise<ReturnType<typeof getPermsByTaskService>>
+) => {
+  const r = await getRethink()
+  const res = await (
+    r
+      .table('Task')
+      .getAll(teamId, {index: 'teamId'})
+      .filter((row) => row('integration').ne(null).default(false).and(row('userId').eq(userId)))
+      .group((row) => [
+        row('userId').default(null),
+        row('integration')('service'),
+        row('integration')('projectKey').default(null),
+        row('integration')('cloudId').default(null),
+        row('integration')('nameWithOwner').default(null)
+      ]) as any
+  )
+    .max('createdAt')('createdAt')
+    .ungroup()
+    .orderBy(r.desc('reduction'))
+    .map((row) => ({
+      userId: row('group')(0),
+      service: row('group')(1),
+      projectKey: row('group')(2),
+      cloudId: row('group')(3),
+      nameWithOwner: row('group')(4),
+      lastUsedAt: row('reduction')
+    }))
+    .run()
+
+  const threeMonthsAgo = new Date(Date.now() - ms('90d'))
+  return res
+    .filter(
+      (res) =>
+        permLookup[res.service] &&
+        (res.service !== 'jira' || res.projectKey) && // jira integrations are making it through that don't have a projectKey
+        res.lastUsedAt > threeMonthsAgo
```

Comment:
If a task was integrated with a project or repo within the past 3 months, it's probably more relevant than other projects & repos and should show up first in the dropdown menu